crates/services/p2p/src/service.rs

```diff
@@ -246,6 +263,23 @@ where
                                     );
                                 });
                             }
+                            RequestMessage::Transactions(block_id) => {
+                                let db = self.db.clone();
+                                let request_sender = self.shared.request_sender.clone();
+
+                                tokio::spawn(async move {
```

Comment:
Just noting here and above are spawns that are not ever joined. This is equivalent to an unbounded channel. There's nothing stopping someone spamming (or even legitimately requesting data) until the machine runs out of memory. I really think we want to maintain backpressure in all our production code.

Reply:
Let's avoid these spawns for now if possible to avoid backpressure issues. Improving concurrency can come later.

Reply:
this has been handled in @freesig's PR to this branch, but we need a long term better solution